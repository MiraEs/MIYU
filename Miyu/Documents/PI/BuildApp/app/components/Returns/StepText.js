'use strict';
import React, {
	PureComponent,
} from 'react';
import PropTypes from 'prop-types';
import {
	StyleSheet,
	View,
	ViewPropTypes,
} from 'react-native';
import {
	Text,
} from 'BuildLibrary';
import styles from '../../lib/styles';

const componentStyles = StyleSheet.create({
	stepWrapper: {
		backgroundColor: styles.colors.white,
		padding: styles.measurements.gridSpace1,
	},
	stepText: {
		alignSelf: 'flex-end',
	},
});

export default class StepText extends PureComponent {
	render() {
		const {
			currentStep,
			maxStep,
			style,
		} = this.props;

		return (
			<View style={[componentStyles.stepWrapper, style]}>
				<Text style={componentStyles.stepText}>Step {currentStep} of {maxStep}</Text>
			</View>
		);
	}
}

StepText.propTypes = {
	currentStep: PropTypes.number.isRequired,
	maxStep: PropTypes.number.isRequired,
	style: ViewPropTypes.style,
};
