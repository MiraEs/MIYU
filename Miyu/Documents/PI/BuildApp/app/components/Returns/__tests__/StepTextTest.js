jest.mock('BuildNative');
jest.mock('BuildLibrary');
jest.mock('react-native');

import StepText from '../StepText';
import React from 'react';
import renderer from 'react-test-renderer';

const defaultProps = {
	currentStep: 1,
	maxStep: 3,
};

describe('StepText component', () => {
	it('it should render', () => {
		const tree = renderer.create(
			<StepText {...defaultProps} />
		);
		expect(tree.toJSON()).toMatchSnapshot();
	});
});
