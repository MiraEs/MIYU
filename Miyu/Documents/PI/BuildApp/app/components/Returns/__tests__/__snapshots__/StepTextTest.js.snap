// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepText component it should render 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#FFFFFF",
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      Object {
        "alignSelf": "flex-end",
      }
    }
  >
    Step 
    1
     of 
    3
  </Text>
</View>
`;
